### DISCUSSION TITLE
Embed Python Interpreter with PyO3 and optionally support multiple version?

### DISCUSSION THREAD

**Discussion by tisonkun** [PyO3/pyo3#4102] [Questions]
@davidhewitt @messense (in case it's more comfortable to discuss in your Discussions forum, or we can continue on the PR linked below)

We are trying to ship the Python script function with PyO3 primarily here. And since statically link to libpython can be complicated, I'm investigating the way to dynamically link to libpython.

So far, it works well and we can see the CI all passed. But this requires a `libpython3.8.so` can be linked in the system.

Thus, I have two questions:

1. How can I make this binary work well with all the libpython >= 3.8?
2. How can I disable the related functionalities if libpython3.so is not found?

This refers to:

* https://github.com/PyO3/pyo3/issues/416
* https://pyo3.rs/v0.21.2/building-and-distribution

_Originally posted by @tisonkun in https://github.com/GreptimeTeam/greptimedb/pull/3749#discussion_r1572249862_

**Comment by davidhewitt** [comment]
Sorry for the delay. I commented in https://github.com/GreptimeTeam/greptimedb/pull/3749
